src/app-layer-parser.c

```diff
@@ -914,6 +944,7 @@ void AppLayerParserTransactionsCleanup(Flow *f)
     bool skipped = false;
     const bool is_unidir =
             AppLayerParserGetOptionFlags(f->protomap, f->alproto) & APP_LAYER_PARSER_OPT_UNIDIR_TXS;
+    uint64_t nbtx = 0;
```

Comment:
I guess we could avoid the double loop if we'd have a proper way to get an actual tx cnt from the parser (instead of the misnomed call to get the max id)